app/src/main/java/org/oppia/android/app/walkthrough/topiclist/WalkthroughTopicListFragmentPresenter.kt

```diff
@@ -105,6 +126,18 @@ class WalkthroughTopicListFragmentPresenter @Inject constructor(
     VIEW_TYPE_TOPIC
   }
 
+  fun changeProgressBarVisibility(visibility: Boolean) {
+    walkthroughVerticalScrollListener.changeProgressBarVisibility(visibility)
+    val recyclerviewHeaderText = binding.walkthroughTopicRecyclerView.getChildAt(0)
+    if (recyclerviewHeaderText.id == R.id.walkthrough_topic_header_text_view) {
+      if (!visibility) {
+        recyclerviewHeaderText.visibility = View.INVISIBLE
```

Comment:
Okay, so this is the cause of the failure. Whenever we are changing the progressBar visibility at the same time we are also dealing with the visibility of inner header text and for that, we need the headerText to be actually there. So, that's why adding some delay will fix this issue.

Reply:
Lets meet and discuss in this in detail.